### DISCUSSION TITLE
test_nearest_method and test_class are trying to use a command line that is too long

### DISCUSSION THREAD

**Discussion by phturb** [mfussenegger/nvim-jdtls#305] [Q&A] [answered]
I was able to successfully install `nvim-jdtls` but when I try to run a jUnit test in one of my test class I'm getting an error. I had a similar problem when I used IDEA intelliJ but was able to fix it with a configuration.

Command on an annotated @Test method : `:lua require'jdtls'.test_nearest_method()`

The problem is : 
```
ERROR - Error on launch: Failed to launch debuggee in terminal. The possible reason is the command line too long. More details: com.sun.jdi.connect.TransportTimeoutException
```

This could be fixed if I were able to use the `shortenCommandLine` argument when launching the tests. The only problem is I can't seems to find a way to do it. I'm not able to find any documentation on how to add argument to these 'jdtls' command. (`test_class` return the same type of problem).

I've tried :
`:lua require'jdtls'.test_nearest_method({shortenCommandLine: "jarmanifest"})`
`:lua require'jdtls'.test_nearest_method({shortenCommandline: "argfile"})`

But when I look at the logs, I can still see my extremely long command line to launch my test.

Has anyone faced this problem before ? If so, how did you fixed it. It's the only missing step before I can fully transfer to an `nvim` development workflow.

**Comment by mfussenegger** [comment, marked as answer]
Can you try with:

```lua
:lua require'jdtls'.test_nearest_method({ config_overrides = { shortenCommandLine: "jarmanifest"} })
```

Or

```lua
:lua require'jdtls'.test_nearest_method({ config_overrides = { shortenCommandLine: "argfile"} })
```

**Reply by phturb** [reply]
Thanks this works to resolve the command too long problem. I still have to resolve why it doesn't find any tests, but this is probably due to some other factor.

This said, these information would be good to add to the readme or doc.

**Reply by mfussenegger** [reply]
`:help jdtls.dap.test_nearest_method` documents the `opts` parameter as `JdtTestOpts` and `:help JdtTestOpts` shows:

```
    Fields: ~
        {config}            (nil|table)         Skeleton used for the |dap-configuration|
        {config_overrides}  (nil|JdtDapConfig)  Overrides for the |dap-configuration|, see |JdtDapConfig|
        {until_error}       (number|nil)        Number of times the test should be repeated if it doesn't fail
        {after_test}        (nil|function)      Callback triggered after test run
```

I guess the documentation for `JdtDapConfig` could mention https://github.com/Microsoft/vscode-java-debug#options
